test/DepartmentService.Business.UnitTests/Commands/Department/EditDepartmentCommandTests.cs

```diff
@@ -0,0 +1,245 @@
+      .Returns(_dbRequest);
+
+    _autoMocker
+      .Setup<IDepartmentRepository, Task<bool>>(x => x.EditAsync(It.IsAny<Guid>(), It.IsAny<JsonPatchDocument<DbDepartment>>()))
+      .ReturnsAsync(true);
+  }
+
+  [Test]
+  public async Task ShouldReturnFailedResponseWhenUserHasNotRightAsync()
+  {
+    _autoMocker
+     .Setup<IAccessValidator, Task<bool>>(x => x.HasRightsAsync(Rights.AddEditRemoveDepartments))
+     .ReturnsAsync(false);
+
+    OperationResultResponse<bool> expectedResponse = new()
+    {
+      Errors = new List<string> { "Not enough rights." }
+    };
+
+    SerializerAssert.AreEqual(expectedResponse, await _command.ExecuteAsync(Guid.NewGuid(), _request));
+
+    _autoMocker.Verify<IAccessValidator>(
+      x => x.HasRightsAsync(It.IsAny<int>()),
+      Times.Once);
+
+    _autoMocker.Verify<IEditDepartmentRequestValidator>(
+      x => x.ValidateAsync(It.IsAny<ValueTuple<Guid, JsonPatchDocument<EditDepartmentRequest>>>(), It.IsAny<CancellationToken>()),
+      Times.Never);
+
+    _autoMocker.Verify<IPatchDbDepartmentMapper>(
+      x => x.Map(It.IsAny<JsonPatchDocument<EditDepartmentRequest>>()),
+      Times.Never);
+
+    _autoMocker.Verify<IDepartmentRepository>(
+      x => x.EditAsync(It.IsAny<Guid>(), It.IsAny<JsonPatchDocument<DbDepartment>>()),
+      Times.Never);
+  }
+
+  [Test]
+  public async Task ShouldReturnFailedResponseWhenValidationIsFailedAsync()
+  {
+    _autoMocker
+      .Setup<IEditDepartmentRequestValidator, Task<ValidationResult>>(x => x.ValidateAsync(It.IsAny<ValueTuple<Guid, JsonPatchDocument<EditDepartmentRequest>>>(), default))
+      .ReturnsAsync(new ValidationResult( new List<ValidationFailure>() { new ValidationFailure() } ));
+
+    OperationResultResponse<bool> expectedResponse = new()
+    {
+      Errors = new List<string> { "Request is not correct." }
+    };
+
+    SerializerAssert.AreEqual(expectedResponse, await _command.ExecuteAsync(Guid.NewGuid(), _request));
+
+    _autoMocker.Verify<IAccessValidator>(
+      x => x.HasRightsAsync(It.IsAny<int>()),
+      Times.Once);
+
+    _autoMocker.Verify<IEditDepartmentRequestValidator>(
+      x => x.ValidateAsync(It.IsAny<ValueTuple<Guid, JsonPatchDocument<EditDepartmentRequest>>>(), It.IsAny<CancellationToken>()),
+      Times.Once);
+
+    _autoMocker.Verify<IPatchDbDepartmentMapper>(
+      x => x.Map(It.IsAny<JsonPatchDocument<EditDepartmentRequest>>()),
+      Times.Never);
+
+    _autoMocker.Verify<IDepartmentRepository>(
+      x => x.EditAsync(It.IsAny<Guid>(), It.IsAny<JsonPatchDocument<DbDepartment>>()),
+      Times.Never);
+  }
+
+  [Test]
+  public async Task ShouldEditDepartmentSuccesfullAsync()
+  {
+    OperationResultResponse<bool> expectedResponse = new()
+    {
+      Body = true
+    };
+
+    SerializerAssert.AreEqual(expectedResponse, await _command.ExecuteAsync(Guid.NewGuid(), _request));
+
+    _autoMocker.Verify<IAccessValidator>(
```

Comment:
May be add Verify method? Like in others services.

Reply:
Done